Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -105,29 +105,78 @@ public IEnumerable<ParticipantStatusExtendedViewModel> GetParticipantStatusHisto
-               var teams = context.ParticipantTeam
-                        .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2 )
-                            .Join(context.Team.Where(t => t.StatusID != 0),
-                                pt => pt.TeamID,
-                                t => t.ID,
-                                (pt, t) => new TeamExtendedViewModel
-                                {
-                                    ID = t.ID,
-                                    Activity = context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
-                                    Name = t.Name,
-                                    Status = context.TeamStatus
-                                                .FirstOrDefault(ts => ts.ID == t.StatusID)
-                                                .Description,
-                                    Logo = t.Logo,
-                                    TeamRecord = context.SeriesTeam
-                                        .Include(st => st.Team)
-                                        .Where(st => st.TeamID == t.ID)
-                                        .Select(st => (TeamRecordViewModel)st)
-                                
-                                });
+        var teams = context.ParticipantTeam
+                 .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2)
+                     .Join(context.Team.Where(t => t.StatusID != 0),
+                         pt => pt.TeamID,
+                         t => t.ID,
+                         (pt, t) => new TeamExtendedViewModel
+                         {
+                             ID = t.ID,
+                             Activity = context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
+                             Name = t.Name,
+                             Status = context.TeamStatus
+                                         .FirstOrDefault(ts => ts.ID == t.StatusID)
+                                         .Description,
+                             Logo = t.Logo,
+                             TeamRecord = context.SeriesTeam
+                                 .Include(st => st.Team)
+                                 .Where(st => st.TeamID == t.ID)
+                                 .Select(st => (TeamRecordViewModel)st)
+
+                         });
         return teams;
     }
 
+    public IEnumerable<MatchExtendedViewModel> GetParticipantMatches(string username)
+    {
+        var matches = context.ParticipantTeam
+            .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2)
+                .Join(context.MatchTeam.Where(mt => mt.StatusID != 0 && mt.Match.StatusID != 0)
+                                        .Include(mt => mt.Match)
+                                            .ThenInclude(m => m.Series)
+                                                .ThenInclude(s => s.Activity)
```

Comment:
According to the documentation (see the end of this section: https://learn.microsoft.com/en-us/ef/core/querying/related-data/eager#including-multiple-levels), you can do this chain of includes with a single call instead then:
```suggestion
                                        .Include(mt => mt.Match.Series.Activity)
```

The docs are not totally clear but I believe that will include all levels of the hierarchy.

Reply:
I can try this out when I regain stable access to the vms